### ISSUE TITLE
For stale read request, we can retry DataIsNotReady replica by replica-read

### ISSUE THREAD

**Issue by crazycs520** [tikv/client-go#1175]
# Issue

### 1. Minimal reproduce step

In the following case, suppose the region leader is in `store1`.

```go
	ca = replicaSelectorAccessPathCase{
		reqType:   tikvrpc.CmdGet,
		readType:  kv.ReplicaReadMixed,
		staleRead: true,
		label:     &metapb.StoreLabel{Key: "id", Value: "2"},
		accessErr: []RegionErrorType{DataIsNotReadyErr, ServerIsBusyErr, ServerIsBusyErr, ServerIsBusyErr},
```

### 2. What did you expect to see?

```go
			accessPath: []string{
				"{addr: store2, replica-read: false, stale-read: true}",
				"{addr: store1, replica-read: false, stale-read: false}", // try leader with leader read.
				"{addr: store2, replica-read: true, stale-read: false}",
				"{addr: store3, replica-read: true, stale-read: false}",
			},
```

### 3. What did you see instead?

```go
			accessPath: []string{
				"{addr: store2, replica-read: false, stale-read: true}",
				"{addr: store1, replica-read: false, stale-read: false}", // try leader with leader read.
				"{addr: store3, replica-read: true, stale-read: false}",
```

Currently, the replica selector logic is:

1. try store2 by stale-read, since store2 has match label, but got `DataIsNotReadyErr`.
2. retry leader in store1, but got `ServerIsBusyErr`.
3. retry store3, but got `ServerIsBusyErr` too.
4. failed, since no replica available.

The problem is, store2 is available, it's meet `DataIsNotReadyErr`, and we can use replica-read to retry store2 again.

**Comment by crazycs520** [comment]
should fix by https://github.com/crazycs520/client-go/commit/2d31d6f9b8f8f75ca4c41f9aab8c3f221d6c0cb9

**Comment by crazycs520** [comment]
fix by #1181
